app/models/website_resolver.rb

```diff
@@ -0,0 +1,58 @@
+class WebsiteResolver
+  attr_reader :results
+
+  def self.resolve(url)
+    resolver = new(url)
+    resolver.resolve
+    resolver
+  end
+
+  def initialize(url)
+    @url = url
+    @results = []
+  end
+
+  def resolve
+    resolve_url if @url
+  end
+
+  def resolved_url
+    @results.last&.fetch :content
+  end
+
+  private
+
+  def resolve_url
+    add_protocol unless protocol?
+    response = connection.get(@url)
+    add_result(response.env)
+  rescue Faraday::Error
+    @results = []
+  end
+
+  def protocol?
+    %r{http(s?)://}.match @url
+  end
+
+  def add_protocol
+    @url = "http://#{@url}"
+  end
+
+  def connection
+    options = { callback: method(:callback) }
```

Comment:
Or maybe a better name? I'll try calling that local `redirect_options` and see if you like that better.